benchmarking/process_json_to_csv.jl

```diff
@@ -50,116 +51,91 @@ import FirstOrderLp
-    NaN,  # l2_primal_variable
-    NaN,  # l_inf_dual_variable
-    FirstOrderLp.TERMINATION_REASON_UNSPECIFIED,  # termination_reason
-    0,  # iteration_count
-    NaN,  # cumulative_kkt_matrix_passes
-    NaN,  # solve_time_sec
-  )
-end
-
-function set_matching_fields(to, from)
-  for field_name in intersect(fieldnames(typeof(to)), fieldnames(typeof(from)))
-    setfield!(to, field_name, getfield(from, field_name))
+function copy_fields_to_dict(to::Dict, from, fields::Vector{Symbol})
+  for fieldname in fields
+    to[fieldname] = getfield(from, fieldname)
   end
+  return
 end
 
-function solve_log_to_csv_row(
-  log::FirstOrderLp.SolveLog,
-  experiment_name::String,
-)::CsvRow
-  row = CsvRow()
-  row.experiment_name = experiment_name
-  set_matching_fields(row, log)
-  row.cumulative_kkt_matrix_passes =
+const SOLVE_LOG_FIELDS_TO_COPY =
+  [:instance_name, :termination_reason, :iteration_count, :solve_time_sec]
+
+const CONVERGENCE_INFORMATION_FIELDS_TO_COPY = [
+  :primal_objective,
+  :dual_objective,
+  :relative_optimality_gap,
+  :l2_primal_residual,
+  :l_inf_primal_residual,
+  :l2_dual_residual,
+  :l_inf_dual_residual,
+  :relative_l2_primal_residual,
+  :relative_l_inf_primal_residual,
+  :relative_l2_dual_residual,
+  :relative_l_inf_dual_residual,
+  :l_inf_primal_variable,
+  :l2_primal_variable,
+  :l_inf_dual_variable,
+]
+
+function solve_log_to_dict(log::FirstOrderLp.SolveLog)::Dict
+  result = Dict()
+  copy_fields_to_dict(result, log, SOLVE_LOG_FIELDS_TO_COPY)
+  result[:cumulative_kkt_matrix_passes] =
     log.solution_stats.cumulative_kkt_matrix_passes
 
   point_type = log.solution_type
   # TODO: This doesn't properly handle the case of infeasibility certificates,
   # whose stats are in log.solution_stats.infeasibility_information.
   for convergence_information in log.solution_stats.convergence_information
     if convergence_information.candidate_type == point_type
-      set_matching_fields(row, convergence_information)
+      copy_fields_to_dict(
+        result,
+        convergence_information,
+        CONVERGENCE_INFORMATION_FIELDS_TO_COPY,
+      )
       break
     end
   end
-  return row
-end
-
-function rows_to_dataframe(rows::Vector{CsvRow})
-  return DataFrame(; StructArrays.components(StructArrays.StructArray(rows))...)
+  return result
 end
 
-struct DatasetNameAndLocation
-  name::String
+struct DatasetConfigAndLocation
+  # Keys must match up with config_labels.
+  config::Dict{String,String}
   logs_directory::String
 end
 
 struct DatasetList
-  datasets::Vector{DatasetNameAndLocation}
+  datasets::Vector{DatasetConfigAndLocation}
+  config_labels::Vector{String}
 end
 
-StructTypes.StructType(::Type{DatasetNameAndLocation}) = StructTypes.Struct()
+StructTypes.StructType(::Type{DatasetConfigAndLocation}) = StructTypes.Struct()
 StructTypes.StructType(::Type{DatasetList}) = StructTypes.Struct()
 
 function read_dataset(dataset_list::DatasetList)::DataFrame
-  rows = CsvRow[]
+  rows = Dict[]
```

Comment:
Optional: to me, it seems slightly more direct to have
   rows = NamedTuple[]
then use 
   push!(rows, NamedTuple(dict))
and finally just
   return DataFrame(rows)
instead of accumulating a list of Dict and then applying a map to convert them to NamedTuple

Reply:
Done